### ISSUE TITLE
Is there some way to check if the information is getting published from kamon to statsd

### ISSUE THREAD

**Issue by murphp15** [kamon-io/docker-grafana-graphite#78]
Hi,
I am very new to kamon statsd, graphite and even Akka.
I am not seeing my metrics in graphana. Is there anyway for meto make sure that my information is being published correctly? Could I set it so Kamon will publish an error if it can't find a statsd server to publish to?

**Comment by vvass** [comment]
I am in the same boat. I am reviewing things right now but cannot figure it out. I believe it had todo with me being new. I know that I am supposed to use the kamon plugin for sbt and there is also a kamon-statsd dependency. The only thing that I don't really get is how to configure all that in applicaion.conf. I am working on it. If get something I will post it here.

**Comment by murphp15** [comment]
@vvass Any luck? I think for me the issue was that I had not mapped the docker port correctly

**Comment by vvass** [comment]
I am still looking at it. I am following some of the examples in the Kamon directories that use kamon statsd (https://github.com/kamon-io/Kamon/tree/master/kamon-examples/kamon-play-example). I am running everything correctly because I was able to perform a few tests. For instance, from inside the container, I issued this command (echo -n "example.statsd.counter.changed:200|c" | nc -w 1 -u 127.0.0.1 8125). Then I sow the counter as an option when creating a singlestat. So I know that it is listening. The next thing that I am trying to do is configure the application.conf file to work. So far I have;

`kamon{

  metric {
    tick-interval = 1 second
  }

  statsd {
    simple-metric-key-generator{
      application="scrbr"
    }
    # Hostname and port in which your StatsD is running. Remember that StatsD packets are sent using UDP and
    # setting unreachable hosts and/or not open ports wont be warned by the Kamon, your data wont go anywhere.
    hostname = "192.168.99.100"
    port = 8125

    # Interval between metrics data flushes to StatsD. It's value must be equal or greater than the
    # kamon.metrics.tick-interval setting.
    flush-interval = 1 second

    # Max packet size for UDP metrics data sent to StatsD.
    max-packet-size = 1024 bytes


    modules {
      kamon-statsd.auto-start = yes
      kamon-log-reporter.auto-start = yes
      kamon-system-metrics.auto-start = no
      kamon-annotation.auto-start = yes
    }

  }
}
` 

As you can see I am trying to match the correct docker machine ip. If I was using the new version of docker I believe I could just point it to 127.0.0.1. I am still attempting to make a connection when running my application. More to come.

**Comment by vvass** [comment]
I am still looking at it. I am following some of the examples in the Kamon directories that use kamon statsd (https://github.com/kamon-io/Kamon/tree/master/kamon-examples/kamon-play-example). I am running everything correctly because I was able to perform a few tests. For instance, from inside the container, I issued this command (echo -n "example.statsd.counter.changed:200|c" | nc -w 1 -u 127.0.0.1 8125). Then I sow the counter as an option when creating a singlestat. So I know that it is listening. The next thing that I am trying to do is configure the application.conf file to work. So far I have;

`kamon{

  metric {
    tick-interval = 1 second
  }

  statsd {
    simple-metric-key-generator{
      application="scrbr"
    }
    # Hostname and port in which your StatsD is running. Remember that StatsD packets are sent using UDP and
    # setting unreachable hosts and/or not open ports wont be warned by the Kamon, your data wont go anywhere.
    hostname = "192.168.99.100"
    port = 8125

    # Interval between metrics data flushes to StatsD. It's value must be equal or greater than the
    # kamon.metrics.tick-interval setting.
    flush-interval = 1 second

    # Max packet size for UDP metrics data sent to StatsD.
    max-packet-size = 1024 bytes


    modules {
      kamon-statsd.auto-start = yes
      kamon-log-reporter.auto-start = yes
      kamon-system-metrics.auto-start = no
      kamon-annotation.auto-start = yes
    }

  }
}
` 

As you can see I am trying to match the correct docker machine ip. If I was using the new version of docker I believe I could just point it to 127.0.0.1. I am still attempting to make a connection when running my application. More to come.

**Comment by ivantopo** [comment]
are you running the play example inside a container as well? and, are you publishing the container port also on port 8125?

**Comment by vvass** [comment]
Yes to publishing the container port on 8125. I have that covered. But I didn't think to try and run the application inside the container. I am going to try that next.

**Comment by ivantopo** [comment]
I wouldn't recommend running the app inside the same container, I was rather asking if you had the application running in a container next to the grafana-graphite container, that might mess up a little the IP you need to use... also, what OS are you running in?

**Comment by vvass** [comment]
Oh, yeah, good point. I am running on Mac. I just moved to using Docker for Mac. I am migrating my images and containers right now to the new Docker setup.

**Comment by ivantopo** [comment]
it has been long since last time I ran docker on a Mac, so I don't know if they have some port redirects being setup or even if they are still running docker inside of VirtualBox, what IP are you using to get into the grafana web interface?

**Comment by vvass** [comment]
They were running Docker inside of Virtual box but recently the moved away from doing that on Mac. Now the use HyperKit. This makes it cleaner. It is almost like running off Linux. There is no more machine IP addresses. It is local, in a sense. 

Before I was using 192.168.99.100 but now I should be able todo the same with 127.0.0.1. I have to run the containers and make sure but the new docker should work this way now. The ports should be fine too. I am going to double check using the previous test I tried from inside the container.

**Comment by vvass** [comment]
I just checked. I still have to use the IP address.

**Comment by ivantopo** [comment]
so, if you go to http://192.168.99.100/ you are seeing the grafana interface there? also, are you super sure that you are publishing the port as UDP? it would be nice if you can share how are you starting the container and the output of docker inspect on the container.

**Comment by vvass** [comment]
Sure, I will attach the output from inspect, but looking at it from an eagles view it seems that it is right:
`"Config": {
            "Hostname": "cbc9743282ac",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "80/tcp": {},
                "81/tcp": {},
                "8125/udp": {},
                "8126/tcp": {}
            },`
[tmp.txt](https://github.com/kamon-io/docker-grafana-graphite/files/693721/tmp.txt)

**Comment by vvass** [comment]
I am using this container project --> https://github.com/kamon-io/docker-grafana-graphite

**Comment by vvass** [comment]
Again, I double checked everything. I am complete convinced that the issue I have is with the way I have set up my configuration files for my spray application. Thank you ivantopo but until i get that right it wont work for me.

**Comment by vvass** [comment]
This is my config in case anyone thinks they can help.
[tmp.txt](https://github.com/kamon-io/docker-grafana-graphite/files/694787/tmp.txt)

**Comment by ivantopo** [comment]
a couple comments on your config:

1. You don't have any filters in your configuration, maybe you need to add some include rules depending on what tools you are using in your application.
2. I wouldn't recommend using 1 second tick interval, it's unnecessary overhead and StatsD works on 10 seconds interval by default anyways.
3. Take a look at this section from your config file:

>     subscriptions {
>       histogram       = [ "*" ]
>       min-max-counter = [ "*" ]
>       gauge           = [ "*" ]
>       counter         = [ "**" ]
>       trace           = [ "*" ]
>       trace-segment   = [ "*" ]
>       akka-actor      = [ "*" ]
>       akka-dispatcher = [ "*" ]
>       akka-router     = [ "*" ]
>       system-metric   = [ "*" ]
>       http-server     = [ "*" ]
>     }

In all those places where you have a single star `*`, you are not going to match any metrics that look like `system/actor`, that's specially bad if you are using Akka because pretty much all entities have a path-like structure. Try to use `**` on the subscriptions for the StatsD module.

4. Ensure you are using the very latest version of docker for mac, according to this: https://docs.docker.com/docker-for-mac/#/step-3-explore-the-application-and-run-examples, all ports should be forwarded to your localhost, meaning that pointing kamon-statsd to 127.0.0.1:8125 should work nicely.

Sadly there are some details about running Docker on OSX that I can't help with, but hopefully you will find a way and share your solution here, regards!

**Comment by vvass** [comment]
Thank @ivantopo !!! Things are looking a lot more promising. I finally started getting something that I was not getting before. 

>10:25:46 INFO [ost-startStop-1] [Kamon                    ] - Initializing Kamon...
>10:25:46 INFO [ost-startStop-1] [Kamon                    ] - Trying to load kamon-autoweave...
>[DEBUG] [01/10/2017 10:25:47.362] [localhost-startStop-1] [EventStream(akka://kamon)] logger log1-Logging$DefaultLogger started
>[DEBUG] [01/10/2017 10:25:47.363] [localhost-startStop-1] [EventStream(akka://kamon)] Default Loggers started
>[INFO] [01/10/2017 10:25:47.416] [localhost-startStop-1] [StatsDExtension(akka://kamon)] Starting the Kamon(StatsD) extension
>[INFO] [01/10/2017 10:25:47.461] [localhost-startStop-1] [LogReporterExtension(akka://kamon)] Starting the Kamon(LogReporter) extension
>[INFO] [01/10/2017 10:25:47.475] [localhost-startStop-1] [SystemMetricsExtension(akka://kamon)] Starting the Kamon(SystemMetrics) extension
>Jan 10, 2017 10:25:47 AM kamon.sigar.SigarProvisioner provision
>INFO: Sigar library provisioned: /Users/vvass/Documents/Development/scrbr/native/libsigar-universal64-macosx.dylib
>10:25:47 INFO [lt-dispatcher-3] [a.e.s.Slf4jLogger        ] - Slf4jLogger started
>[DEBUG] [01/10/2017 10:25:47.877] [kamon-akka.actor.default-dispatcher-9] [EventStream] shutting down: StandardOutLogger started

Here is my new config file: [tmp.txt](https://github.com/kamon-io/docker-grafana-graphite/files/696530/tmp.txt)

Now the only thing I am questioning is the line >[DEBUG] [01/10/2017 10:25:47.877] [kamon-akka.actor.default-dispatcher-9] [EventStream] shutting down: StandardOutLogger started

Do you know if that should be there?

**Comment by vvass** [comment]
At the end of it all I figured out that my spray and akka application, or at least the way I set it up doesn't chive well with kamon and aspectj. So I have decided to move off spray (they are moving to akka-http anyway) and redo my application with play framework. The above configuration does work but my project is to complicated. Hopefully next iteration it will be better. I hope that the examples and conversations above help others. Thank you!
